src/ostorlab/runtimes/definitions.py

```diff
@@ -38,7 +50,7 @@ class AgentSettings:
     bus_management_url: Optional[str] = ""
     bus_vhost: Optional[str] = ""
     args: List[defintions.Arg] = dataclasses.field(default_factory=list)
-    constraints: List[str] = dataclasses.field(default_factory=list)
+    constraints: Optional[List[str]] = dataclasses.field(default_factory=list)
```

Comment:
```suggestion
    constraints: list[str] | None = dataclasses.field(default_factory=list)
```

Reply:
the workflow will fail as this is not supported in python3.9